components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6676,6 +6694,84 @@ where
         self.schedule_tick(PeerTick::ReportBuckets)
     }
 
+    /// Check whether the peer should send a request to fetch the committed
+    /// index from the leader.
+    fn try_to_fetch_committed_index(&mut self) {
+        // Already completed, skip.
+        if self.fsm.peer.busy_on_apply.is_none()
+            || self.fsm.peer.last_leader_committed_idx.is_some()
+        {
+            return;
+        }
+        // As the leader of the region will have the accurate and latest commit index,
+        // we can sync it to the peer who wanna it.
+        let leader_id = self.fsm.peer.leader_id();
+        let leader = self.fsm.peer.get_peer_from_cache(leader_id);
+        if let Some(leader) = leader {
+            let mut msg = ExtraMessage::default();
+            msg.set_type(ExtraMessageType::MsgFetchCommittedIndexRequest);
+            msg.set_index(RAFT_INIT_LOG_INDEX);
+            self.fsm
+                .peer
+                .send_extra_message(msg, &mut self.ctx.trans, &leader);
+            debug!(
+                "try to fetch committed index from leader";
+                "region_id" => self.region_id(),
+                "peer_id" => self.fsm.peer_id()
+            );
+        }
+    }
+
+    /// Handle the request of the committed index from the peer.
+    ///
+    /// Normally, the handler should be follower / learner of this region.
+    fn on_fetch_committed_index_request(&mut self, from: &metapb::Peer) {
+        let committed_index = self.fsm.peer.get_store().commit_index();
+        let mut resp = ExtraMessage::default();
+        resp.set_type(ExtraMessageType::MsgFetchCommittedIndexResponse);
+        resp.set_index(committed_index);
+        self.fsm
+            .peer
+            .send_extra_message(resp, &mut self.ctx.trans, from);
+        debug!(
+            "responses committed to peer";
+            "region_id" => self.region().get_id(),
+            "from" => from.id,
+            "peer_id" => self.fsm.peer.peer.get_id(),
+            "committed_index" => committed_index,
+        );
+    }
+
+    /// Handle the response of the committed index from the peer.
+    ///
+    /// Normally, the handler should be the leader of this region.
+    fn on_fetch_committed_index_response(&mut self, msg: RaftMessage) {
+        // Peer is not leader, it should extract the committed index from the leader
+        // and update its local `last_leader_committed_idx` if the committed index is
+        // valid.
+        let committed_index = msg.get_extra_msg().get_index();
+        if committed_index <= RAFT_INIT_LOG_INDEX {
```

Comment:
This check seems useless? Could you explain in what condition, can this happen?
BTW, if I understand correctly, if the region is newly created and doesn't have any propose, it's possible that the current committed index is `RAFT_INIT_LOG_INDEX `(if the first empty propose is still not committed).

Reply:
yep, this condition should be updated. It should check whether the message is stale or not after changing the message to the pair `MsgFetchCommittedIndexReq & Response`.